demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +120,154 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
 
     return blobs;
 }
+
+
+void EncoderDecoderCNN::Init(const std::string &model_path, Core & ie, const std::string & deviceName, const cv::Size &new_input_resolution) {
+    // ---------------------------------------------------------------------------------------------------
+    // --------------------------- 0. checking paths -----------------------------------------------------
+    std::string model_path_decoder = model_path;
+    while (model_path_decoder.find("encoder") != std::string::npos)
+        model_path_decoder = model_path_decoder.replace(model_path_decoder.find("encoder"), 7, "decoder");
```

Comment:
Implemented the second option

Reply:
Report an error if one sets wrong `decoder_type` for composite model. The same is for `tr_pt_first`: it would be strange to silently add symbols in front for `EncoderDecoderCNN` when key value says that pad token is not the first symbol in the alphabet.

Replace `setInOutNames()` `Init()` pair with normal constructor for EncoderDecoderCNN: remove `setInOutNames()` define `Init()` as empty.
I'm thinking of doing the same for `text_detection`...

Reply:
Done

Reply:
Report an error if `EncoderDecoderCNN` is successfully built and `FLAGS_tr_pt_first` is false. You know for sure it should have been true. So add `std::string(3, kPadSymbol)` unconditionally before that.

Reply:
Don't require `FLAGS_tr_pt_first` to be true until it is known that the model is composite, which happens after `EncoderDecoderCNN` construction